### ISSUE TITLE
[BUG] -fsyntax-only no longer works

### ISSUE THREAD

**Issue by Viatorus** [compiler-explorer/compiler-explorer#2621]
**Describe the bug**
Running GCC with the compiler option `-fsyntax-only` results in:

> cc1plus: fatal error: cannot open '/dev/null' for writing: Permission denied

But it did work before. I use this flag to speedup syntax checks if I am not interested in assembler outputs (like in https://viatorus.github.io/compile-time-printer/ - which isn't working anymore because of this change).

**To Reproduce**
Add `-fsyntax-only` as compiler option.

**Expected behavior**
Should work without exception.

**Link**
https://gcc.godbolt.org/z/bzaYjnsoG

**Comment by mattgodbolt** [comment]
Thanks! Looks like an easy one to fix, thankfully! Congrats on finding the first issue with our new sandboxing mechanism :)

**Comment by mattgodbolt** [comment]
Repro'd locally. I need to work out what's going on here, because it _looks_ like it should work, but no:

```bash
$ 
nsjail --config etc/nsjail/execute.cfg -- /bin/bash
bash-5.0$ ls -l /dev
total 0
crw-rw-rw- 1 65534 65534 1, 3 Apr 23 08:15 null
crw-rw-rw- 1 65534 65534 1, 9 Apr 23 08:15 random
crw-rw-rw- 1 65534 65534 1, 9 Apr 23 08:15 urandom
crw-rw-rw- 1 65534 65534 1, 5 Apr 23 08:15 zero
bash-5.0$ cat > /dev/null 
bash: /dev/null: Permission denied
```

**Comment by mattgodbolt** [comment]
with debug logging we can see nsjail thinks it's doing the right thing:
```
[D][2021-04-26T13:43:32-0500][1] bool mnt::mountPt(mount_t*, const char*, const char*)():126 Mounting ''/dev/null' -> '/dev/null' flags:MS_BIND|MS_REC|MS_PRIVATE type:'' options:'' dir:false'
```

**Comment by mattgodbolt** [comment]
Aha! the (apparently unnecessary) `/dev` mount "above" makes it read-only.

**Comment by mattgodbolt** [comment]
Closed by 8a2cbebc70a552322260ca7561653ee6ef669e6d

**Comment by Viatorus** [comment]
Awesome! It is working again. Thank you very much for your fast help and also your debugging thriller story. ;)
